lib/ansible/modules/windows/win_dsc.ps1

```diff
@@ -0,0 +1,236 @@
+    }
+    ElseIf ($prop.PropertyType -eq "[string[]]")
+    {
+        #KeyValue is an array of strings
+        [String]$TempKeyValue = $attrib.Item($_)
+        [String[]]$KeyValue = $TempKeyValue.Split(",").Trim()
+
+        $config.Property.Add($key,$KeyValue)
+    }
+    ElseIf ($prop.PropertyType -eq "[UInt32[]]")
+    {
+        #KeyValue is an array of integers
+        [String]$TempKeyValue = $attrib.Item($_)
+        [UInt32[]]$KeyValue = $attrib.Item($_.split(",").Trim())
+        $config.Property.Add($key,$KeyValue)
+    }
+    ElseIf ($prop.PropertyType -eq "[bool]")
+    {
+        if ($attrib.Item($_) -like "true")
+        {
+            [bool]$KeyValue = $true
+        }
+        ElseIf ($attrib.Item($_) -like "false")
+        {
+            [bool]$KeyValue = $false
+        }
+        $config.Property.Add($key,$KeyValue)
+    }
+    ElseIf ($prop.PropertyType -eq "[int]")
+    {
+        [int]$KeyValue = $attrib.Item($_)
+        $config.Property.Add($key,$KeyValue)
+    }
+    ElseIf ($prop.PropertyType -eq "[CimInstance[]]")
+    {
+      #KeyValue is an array of CimInstance
+      [CimInstance[]]$KeyVal = @()
+      [String]$TempKeyValue = $attrib.Item($_)
+      #Need to split on the string }, because some property values have commas in them
+      [String[]]$KeyValueStr = $TempKeyValue -split("},")
+      #Go through each string of properties and create a hash of them
+      foreach($str in $KeyValueStr)
+      {
+        [string[]]$properties = $str.Split("{")[1].Replace("}","").Trim().Split([environment]::NewLine).Trim()
+        $prph = @{}
+        foreach($p in $properties)
+        {
+          $pArr = $p -split "="
+          #if the value can be an int we must convert it to an int
+          if([bool]($pArr[1] -as [int] -is [int]))
+          {
+              $prph.Add($pArr[0].Trim(),$pArr[1].Trim() -as [int])
+          }
+          else
+          {
+              $prph.Add($pArr[0].Trim(),$pArr[1].Trim())
+          }
+        }
+        #create the new CimInstance
+        $cim = New-CimInstance -ClassName $str.Split("{")[0].Trim() -Property $prph -ClientOnly
+        #add the new CimInstance to the array
+        $KeyVal += $cim
+      }
+      $config.Property.Add($key,$KeyVal)
+    }
+  }
+
+try
+{
+    #Defined variables in strictmode
+    $TestError, $TestError = $null
+    $TestResult = Invoke-DscResource @Config -Method Test -ModuleName $Module -ErrorVariable TestError -ErrorAction SilentlyContinue
+    if ($TestError)
+    {
+       throw ($TestError[0].Exception.Message)
+    }
+    ElseIf (($TestResult.InDesiredState) -ne $true)
+    {
+        if ($check_mode -eq $False)
+        {
+            $SetResult = Invoke-DscResource -Method Set @Config -ModuleName $Module -ErrorVariable SetError -ErrorAction SilentlyContinue -WarningAction SilentlyContinue
+            set-attr -obj $result -name "reboot_required" -value $SetResult.RebootRequired
+        }
+
+        Set-Attr $result "changed" $true
```

Comment:
Should we not use the defined "Set-Attr" from Ansible, for the Future?
So Ansible Ps Execution Module (Script Includion) also see the changes in the Result Object/Work with it?
Maybe thats another stuff to discuss on Coding standards for Windows Modules :-)